external-api/src/v1/controllers/estore/eStore.controller.ts

```diff
@@ -130,34 +159,35 @@ export const create = async (req: EstoreRequest, res: Response) => {
         },
       });
 
-      // Step 2: Site exists check
+      if (!inserted) {
+        throw new Error('eStore CRON job log insertion failed');
+      }
+
+      const { insertedId: _id } = inserted as InsertOneResult;
+
+      // Site exists check
       console.info('Initiating eStore site existence check for exporter %s', eStoreData.exporterName);
       const siteExistsResponse: SiteExistsResponse | EstoreErrorResponse = await siteExists(eStoreData.exporterName);
 
       const created = siteExistsResponse?.data?.status === ESTORE_SITE_STATUS.CREATED;
       const provisioning = siteExistsResponse?.data?.status === ESTORE_SITE_STATUS.PROVISIONING;
       const absent = siteExistsResponse?.data?.status === ESTORE_SITE_STATUS.NOT_FOUND;
 
-      // Step 3: Site already exists in eStore
+      // Site already exists in eStore
       if (created) {
         /**
          * Update record-set with the site name.
          * Update `cron-job-logs`
          */
-        await cronJobLogs.updateOne(
-          { _id: { $eq: new ObjectId(_id) } },
-          {
-            $set: {
-              'cron.site.create': {
-                status: ESTORE_CRON_STATUS.COMPLETED,
-                response: siteExistsResponse.data.status,
-                timestamp: getNowAsEpoch(),
-                id: siteExistsResponse.data.siteId,
-              },
-              'cron.site.status': ESTORE_CRON_STATUS.COMPLETED,
-            },
+        await EstoreRepo.updateByDealId(eStoreData?.dealId, {
```

Comment:
`eStoreData?.dealId` is repeated a few times and on line 139 it does not have optional chaining.  Is optional chaining required and could dealId be dereferenced with an if(!dealId) { do something }?

Reply:
Good spot, have updated and referred to the variables from `body`

```typescript
    const { dealId, siteId, facilityIdentifiers, supportingInformation, exporterName, buyerName, dealIdentifier, destinationMarket, riskMarket } = body;
```

Reply:
Think this one has been missed, can be done in the next PR